api/controllers/bulk-docs.js

```diff
@@ -1,18 +1,17 @@
 const auth = require('../auth');
-const serverUtils = require('../server-utils');
 const bulkDocs = require('../services/bulk-docs');
 
 module.exports = {
   bulkDelete: (req, res, next) => {
-    auth.getUserCtx(req, function(err, userCtx) {
-      if (err) {
-        return serverUtils.error(err, req, res);
-      } else if (!auth.isAdmin(userCtx)) {
-        return serverUtils.error({ code: 401 }, req, res);
-      }
-
-      bulkDocs.bulkDelete(req.body.docs, res, { batchSize: 100 })
-        .catch(err => next(err));
-    });
+    return auth.getUserCtx(req)
+      .then(userCtx => {
+        if (!auth.isAdmin(userCtx)) {
+          const error = Error();
```

Comment:
FWIW, and I have no idea if this is better or worse: because types aren't really a thing and we don't care about generating stack traces I've just been doing:

```js
throw {
  code: 123,
  message: 'This is the police speaking! This club is CLOSED!'
};
```
I may be a terrible human though 🐴

Reply:
Oh cool yeah I prefer that style, couldn't think of a nice way to do it, will change 👍